appservice/src/main/java/nu/yona/server/messaging/rest/MessageController.java

```diff
@@ -250,7 +250,7 @@ public MessageDto toResource(MessageDto message)
 
 		private void addRelatedMessageLink(MessageDto message, MessageDto messageResource)
 		{
-			if (message.getRelatedMessageId() != null)
+			if (message.getRelatedMessageId() != 0)
```

Comment:
Shouldn't we use `Optional<Long>` here instead of `long`, as with `repliedMessageId` below?